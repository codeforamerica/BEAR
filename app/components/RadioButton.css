.radioButton {
  -webkit-appearance: none;
  top: 0.6em;
  left: 0.7em;
  height: 22px;
  width: 22px;
  padding: 0.3em !important;
  border: 2px solid #30302f !important;
  border-radius: 50% !important;
}

.radioButton:checked::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -6px;
  margin-left: -6px;
  height: 12px;
  width: 12px;
  border-radius: 50%;
  background-color: #017f6a;
}
