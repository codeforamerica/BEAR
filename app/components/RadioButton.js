import React, { Component } from 'react';
import styles from './RadioButton.css';

type Props = {
  selected: boolean,
  value: string,
  group: string,
  onSelect: (string, string) => void
};

export default class RadioButton extends Component<Props> {
  handleSelect = () => {
    const { onSelect, group, value } = this.props;
    onSelect(group, value);
  };

  render() {
    const { selected, value, group } = this.props;

    return (
      <input
        defaultChecked={selected}
        type="radio"
        className={`${styles.radioButton} radio-button`}
        name={group}
        id={`${value}_${group}`}
        onChange={this.handleSelect}
      />
    );
  }
}
