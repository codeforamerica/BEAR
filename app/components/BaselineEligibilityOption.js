import React, { Component } from 'react';
import RadioButton from './RadioButton';

type Props = {
  codeSection: string,
  selectedOption: string
};

export default class BaselineEligibilityOption extends Component<Props> {
  render() {
    const { codeSection, selectedOption } = this.props;
    return (
      <tr>
        <td>
          <p>{`HS ${codeSection} felonies`}</p>
        </td>
        <td>
          <RadioButton
            selected={selectedOption === 'dismiss'}
            value="dismiss"
            group={codeSection}
          />
        </td>
        <td>
          <RadioButton
            selected={selectedOption === 'reduce'}
            value="reduce"
            group={codeSection}
          />
        </td>
      </tr>
    );
  }
}
