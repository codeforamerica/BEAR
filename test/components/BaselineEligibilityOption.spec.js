import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BaselineEligibilityOption from '../../app/components/BaselineEligibilityOption';
import RadioButton from '../../app/components/RadioButton';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const options = {
    '11357': 'dismiss',
    '11358': 'reduce'
  };
  const component1 = shallow(
    <BaselineEligibilityOption
      codeSection="11357"
      baselineEligibilityOptions={options}
    />
  );

  const component2 = shallow(
    <BaselineEligibilityOption
      codeSection="11358"
      baselineEligibilityOptions={options}
    />
  );
  return {
    component1,
    component2
  };
}

describe('BaselineEligibilityOption component', () => {
  it('if dismissed, reduce should not be selected', () => {
    const { component1 } = setup();
    expect(
      component1.containsAnyMatchingElements([
        <RadioButton selected={true} value="dismiss" />
      ])
    ).toEqual(true);

    expect(
      component1.containsAnyMatchingElements([
        <RadioButton selected={false} value="reduce" />
      ])
    ).toEqual(true);
  });

  it('if reduced, dismiss should not be selected', () => {
    const { component2 } = setup();
    expect(
      component2.containsAnyMatchingElements([
        <RadioButton selected={true} value="reduce" />
      ])
    ).toEqual(true);

    expect(
      component2.containsAnyMatchingElements([
        <RadioButton selected={false} value="dismiss" />
      ])
    ).toEqual(true);
  });

  it('should match exact snapshot', () => {
    const { component1 } = setup();
    const tree = renderer.create(component1).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
