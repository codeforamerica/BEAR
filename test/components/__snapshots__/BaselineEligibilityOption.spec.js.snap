// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaselineEligibilityOption component should match exact snapshot 1`] = `
<tr>
  <td>
    <p>
      H&S ยง undefined
    </p>
  </td>
  <td>
    <input
      className="radioButton radio-button"
      defaultChecked={true}
      id="dismiss_11357(a)"
      name="11357(a)"
      onChange={[Function]}
      type="radio"
    />
  </td>
  <td>
    <input
      className="radioButton radio-button"
      defaultChecked={false}
      id="reduce_11357(a)"
      name="11357(a)"
      onChange={[Function]}
      type="radio"
    />
  </td>
</tr>
`;
